spokestack/nlu/parsers/integer.py

```diff
@@ -0,0 +1,70 @@
+"""
+This module contains the logic to parse integers from NLU results
+"""
+from typing import Any, Dict, Union
+
+from spokestack.nlu.parsers import maps
+from spokestack.nlu.parsers.digits import DIGIT_SPLIT_RE
+
+
+def parse(metadata: Dict[str, Any], raw_value: str) -> Union[int, None]:
+    """ Parser for Integer slots
+
+    Args:
+        metadata (Dict[str, Any]): metadata for the integer slot
+        raw_value (str): value tagged by the model
+
+    Returns:
+        Union[int, None]: integer if parsable, None if invalid
+    """
+    raw_range = metadata.get("range")
+    normalized = raw_value.lower()
+    tokens = DIGIT_SPLIT_RE.split(normalized)
+
+    parsed_values = []
+    for token in tokens:
+        try:
+            parsed = int(token)
+            parsed_values.append(parsed)
+        except ValueError:
+            reduced = _parse_reduce(token, parsed_values)
+            if not reduced:
+                return None
```

Comment:
```suggestion
            if _parse_reduce(token, parsed_values) is None:
                return None
```
not mandatory, just a thought